api/workers/initialize.js

```diff
@@ -0,0 +1,410 @@
+      _getReleaseVersion.bind(null, bag),
+      _generateInitializeEnvs.bind(null, bag),
+      _generateInitializeScript.bind(null, bag),
+      _writeScriptToFile.bind(null, bag),
+      _initializeWorker.bind(null, bag),
+      _post.bind(null, bag)
+    ],
+    function (err) {
+      logger.info(bag.who, 'Completed');
+      if (!bag.skipStatusChange)
+        _setCompleteStatus(bag, err);
+
+      if (err) {
+        // only send a response if we haven't already
+        if (!bag.isResponseSent)
+          respondWithError(bag.res, err);
+        else
+          logger.warn(err);
+      }
+      //TODO: remove tmp file
+    }
+  );
+}
+
+function _checkInputParams(bag, next) {
+  var who = bag.who + '|' + _checkInputParams.name;
+  logger.verbose(who, 'Inside');
+
+  if (!_.has(bag.reqBody, 'address') || _.isEmpty(bag.reqBody.address))
+    return next(
+      new ActErr(who, ActErr.DataNotFound,
+        'Missing body data :address')
+    );
+
+  bag.address = bag.reqBody.address;
+
+  return next();
+}
+
+function _get(bag, next) {
+  var who = bag.who + '|' + _get.name;
+  logger.verbose(who, 'Inside');
+
+  configHandler.get(bag.component,
+    function (err, workers) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.DataNotFound,
+            'Failed to get ' + bag.component, err)
+        );
+
+      if (_.isEmpty(workers))
+        return next(
+          new ActErr(who, ActErr.DataNotFound,
+            'No configuration in database for ' + bag.component)
+        );
+
+      bag.config = _.find(workers,
+        function (worker) {
+          return worker.address === bag.address;
+        }
+      );
+
+      if (_.isEmpty(bag.config))
+        return next(
+          new ActErr(who, ActErr.DataNotFound,
+            'No entry in workers list for ' + bag.address)
+        );
+
+      return next();
+    }
+  );
+}
+
+function _getMaster(bag, next) {
+  var who = bag.who + '|' + _getMaster.name;
+  logger.verbose(who, 'Inside');
+
+  configHandler.get('master',
+    function (err, master) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.DataNotFound,
+            'Failed to get master', err)
+        );
+
+      if (_.isEmpty(master))
+        return next(
+          new ActErr(who, ActErr.DataNotFound,
+            'No configuration in database for master')
+        );
+
+      bag.master = master;
+      return next();
+    }
+  );
+}
+
+function _checkConfig(bag, next) {
+  var who = bag.who + '|' + _checkConfig.name;
+  logger.verbose(who, 'Inside');
+
+  var missingConfigFields = [];
+
+  if (!_.has(bag.config, 'address') || _.isEmpty(bag.config.address))
+    missingConfigFields.push('address');
+  if (!_.has(bag.config, 'port') || !bag.config.port)
+    missingConfigFields.push('port');
+
+  if (missingConfigFields.length)
+    return next(
+      new ActErr(who, ActErr.DataNotFound,
+        'Missing config data: ' + missingConfigFields.join())
```

Comment:
How about "Missing worker config data"?  It might not be clear that it's the worker config and not the master.